gammapy/datasets/map.py

```diff
@@ -2342,47 +2344,255 @@ def to_image(self, name=None):
+            exposure_map : `Map`
+                Gammapy Map containing the exposure data (with energy axis).
+            """
+
+            exposure_map.plot(ax=ax, ls="solid", marker=None, xerr=None)
+            # Dynamically format the y-axis label
+            unit = exposure_map.unit.to_string("latex")  # Convert unit to LaTeX format
+            ax.set_ylabel(f"Exposure [{unit}]")  # Set the formatted y-axis label
+            ax.set_title("Exposure (at FoV center)")
+            ax.set_box_aspect(1)
+
+        def plot_containment_radius(ax, psf):
+            """
+            Plot the PSF containment radius curves. By default, we rely on a built-in
+            Gammapy PSF plotting method if available.
+
+            Parameters
+            ----------
+            ax : `matplotlib.axes.Axes`
+                The matplotlib Axes to plot on.
+            psf : `PSFKernel` or `PSFMap`
+                The Gammapy PSF object.
+            """
+            psf.plot_containment_radius_vs_energy(ax=ax)
+            ax.legend(fontsize="small")
+            ax.set_title("Containment radius (at FoV center)")
+            ax.set_xscale("log")
+            ax.set_yscale("log")
+            ax.set_box_aspect(1)
+
         def plot_mask(ax, mask, **kwargs):
+            """
+            Plot a mask (if provided) on the given axis.
+
+            Parameters
+            ----------
+            ax : `matplotlib.axes.Axes`
+                The matplotlib Axes to plot on.
+            mask : `Map` or None
+                Gammapy map that is interpreted as a mask (boolean or float).
+                If None, no mask is plotted.
+            kwargs : dict
+                Additional keyword arguments forwarded to mask plotting (e.g. alpha, hatches, colors).
+            """
             if mask is not None:
                 mask.plot_mask(ax=ax, **kwargs)
 
-        fig, axes = plt.subplots(
-            ncols=2,
-            nrows=2,
-            subplot_kw={"projection": self._geom.wcs},
-            figsize=figsize,
-            gridspec_kw={"hspace": 0.25, "wspace": 0.1},
+        # Reduce the datasets to 2D if needed
+        countsmapdata = self.counts.reduce_over_axes()
+        npredmapdata = self.npred().reduce_over_axes()
+
+        # Get the corresponding central pixel SpectrumDataset (exposure, edisp, psf)
+        central_pixel = RectangleSkyRegion(
+            self.counts.geom.center_skydir,
+            width=1.01 * self.counts.geom.pixel_scales[0],
+            height=1.01 * self.counts.geom.pixel_scales[1],
+        )
+        central_spectrum_dataset = self.to_spectrum_dataset(central_pixel)
+
+        # Determine plotting limits
+        vmin = npredmapdata.data.min()
+        vmax = npredmapdata.data.max()
+        # Fallback if the map is entirely zero
+        if vmin == 0.0:
+            vmin = np.max([countsmapdata.data.max() * 0.02, countsmapdata.data.min()])
+        if vmax == 0.0:
+            vmax = countsmapdata.data.max()
```

Comment:
In general, one could have the min and max of the npred equal to that of the counts. No?